packages/dd-trace/src/dogstatsd.js

```diff
@@ -48,16 +56,37 @@ class Client {
     }
   }
 
-  _send (address, family, buffer) {
+  _sendHttp (options, address, family, buffer) {
+    request(buffer, options, (err, _, code) => {
+      if (err) {
+        log.error('HTTP error from agent: ' + err.stack)
+        if (err.status) {
+          // Inside this if-block, we have connectivity to the agent, but
+          // we're not getting a 200 from the proxy endpoint. Fall back to
+          // UDP and try again.
+          this._httpOptions = null
+          this._sendUdp(address, family, buffer)
+        }
+      }
+    })
+  }
+
+  _sendUdp (address, family, buffer) {
     const socket = family === 6 ? this._udp6 : this._udp4
 
     log.debug(`Sending to DogStatsD: ${buffer}`)
 
     socket.send(buffer, 0, buffer.length, this._port, address)
   }
 
-  _sendAll (queue, address, family) {
-    queue.forEach((buffer) => this._send(address, family, buffer))
+  _sendAll (queue, address, family, options) {
+    queue.forEach((buffer) => {
+      if (options) {
```

Comment:
This condition is repeated here and in the parent. Why not just call the right method directly?

Reply:
Do you mean like:

```js
if (options) {
  queue.forEach(buffer => this._sendHttp(buffer))
} else {
    queue.forEach(buffer => this._sendUdp(buffer))
}
```

Or do you mean having a `_sendAllHttp` and `_sendAllUdp`?

Reply:
I mean calling the right one in the parent yes. Although looking at it more it does multiple HTTP requests but it should make only 1 since HTTP is much more inefficient compared to UDP.